promise.d.ts

```diff
@@ -78,17 +82,16 @@ export interface Connection extends QueryableAndExecutableBase {
 
 export interface PoolConnection extends Connection {
   release(): void;
-  connection: Connection;
 }
 
-export interface Pool extends Connection {
+export interface Pool extends QueryableAndExecutableBase, Pick<CorePool, 'escape' | 'escapeId' | 'format'> {
```

Comment:
Pool doesn't extend Connection in code, so this better reflects the actual implementation